packages/compass-utils/src/cancellable-promise.ts

```diff
@@ -17,66 +17,26 @@ export function isCancelError(error: any): error is AbortError {
   return error?.name === 'AbortError';
 }
 
-/*
- * Return a promise you can race (just like a timeout from timeouts/promises).
- * It will reject if abortSignal triggers before successSignal
- */
-function abortablePromise(
-  abortSignal: AbortSignal,
-  successSignal: AbortSignal
-) {
-  if (abortSignal.aborted) {
-    return Promise.reject(abortSignal.reason ?? createCancelError());
-  }
-
-  let reject: (reason: unknown) => void;
-
-  const promise = new Promise<never>(function (_resolve, _reject) {
-    reject = _reject;
-  });
-
-  const abort = () => {
-    // if this task aborts it will never succeed, so clean up that event listener
-    // (abortSignal's event handler is already removed due to { once: true })
-    successSignal.removeEventListener('abort', succeed);
-
-    reject(createCancelError());
-  };
-
-  const succeed = () => {
-    // if this task succeeds it will never abort, so clean up that event listener
-    // (successSignal's event handler is already removed due to { once: true })
-    abortSignal.removeEventListener('abort', abort);
-  };
-
-  abortSignal.addEventListener('abort', abort, { once: true });
-  successSignal.addEventListener('abort', succeed, { once: true });
-
-  return promise;
-}
-
-/*
- * We need a promise that will reject as soon as the operation is aborted since
- * closing the cursor isn't enough to immediately make the cursor method's
- * promise reject.
- */
 export async function raceWithAbort<T>(
   promise: Promise<T>,
   signal: AbortSignal
 ): Promise<T> {
   if (signal.aborted) {
     return Promise.reject(signal.reason ?? createCancelError());
   }
-  const successController = new AbortController();
-  const abortPromise = abortablePromise(signal, successController.signal);
+
+  let abortListener;
+
+  // We need a promise that will reject as soon as the operation is aborted.
+  const pendingPromise = new Promise<never>((_resolve, reject) => {
+    abortListener = () => reject(createCancelError());
+    signal.addEventListener('abort', abortListener, { once: true });
+  });
+
   try {
-    return await Promise.race([abortPromise, promise]);
+    return await Promise.race([pendingPromise, promise]);
   } finally {
-    if (!signal.aborted) {
-      // either the operation succeeded or it failed because of some error
-      // that's not an abort
-      successController.abort();
-    }
+    abortListener && signal.removeEventListener('abort', abortListener);
```

Comment:
This first part is just to silence TS, right?

Reply:
Yes.